appstore/developer_portal_api.py

```diff
@@ -421,6 +421,108 @@ def delete_screenshot(app_id, platform, screenshot_id):
     db.session.commit()
     return jsonify(success=True, message=f"Deleted screenshot {screenshot_id}", id=screenshot_id, platform=platform)
         
+@devportal_api.route('/app/<app_id>/banners/<platform>', methods=['GET'])
+def get_app_banners(app_id, platform):
+    # Check app exists
+
+    if not is_valid_platform(platform):
+        return jsonify(error=f"Invalid platform: {platform}", e="platform.invalid"), 400  
+
+    try:
+        app = App.query.filter(App.id == app_id).one()
+    except NoResultFound as e:
+        return jsonify(error="Unknown app", e="app.notfound"), 400    
+
+    asset_collection = AssetCollection.query.filter(AssetCollection.app_id == app.id, AssetCollection.platform == platform).one_or_none()
+
+    if asset_collection is None or asset_collection.headers is None:
+        return jsonify([])
+    else:
+        return jsonify(asset_collection.headers)
+
+@devportal_api.route('/app/<app_id>/banners/<platform>', methods=['POST'])
+def new_app_banner(app_id, platform):
+    try:
+        app = App.query.filter(App.id == app_id).one()
+    except NoResultFound as e:
+        return jsonify(error="Unknown app", e="app.notfound"), 400
+
+    # Check we own the app
+    if not is_users_developer_id(app.developer_id):
+        return jsonify(error="You do not have permission to modify that app", e="permission.denied"), 403
+
+    if not is_valid_platform(platform):
+        return jsonify(error=f"Invalid platform: {platform}", e="platform.invalid"), 400  
+
+    asset_collection = AssetCollection.query.filter(AssetCollection.app_id == app.id, AssetCollection.platform == platform).one_or_none()
+
+    # Get the image, this is a single image API
+    if "banner" in request.files:
+        new_image = request.files["banner"]
+    else:
+        return jsonify(error="Missing file: banner", e="banner.missing"), 400
+
+    # Check it's a valid image file
+    if not is_valid_image_file(new_image):
+        return jsonify(error="Illegal image type", e="banner.illegalvalue"), 400
+
+    # Check it's the correct size
+    if not is_valid_image_size(new_image, "banner"):
+        max_w, max_h = get_max_image_dimensions("banner")
+        return jsonify(error="Invalid image size", e="banner.illegaldimensions", message=f"Image should be {max_w}x{max_h}"), 400
+        
+    if asset_collection is None:
+        # With screenshots we create new asset collection
+        # However, if we do that here we can end up with an asset collection for a platform that has no screenshots. So let's fail here and force the user to upload a screenshot for that platform first
+        return jsonify(error="You cannot add a banner for a platform which has no screenshots", e="prerequisite.missing", message="Please add at least one screenshot for the selected platform, then retry the banner upload."), 409
+    else:
+        # Check we don't already have 3 banners in this asset collection
+        if len(asset_collection.headers) > 3:
```

Comment:
What's the limit supposed to be? The comment implies it's three, in which case the comparison should be `>=`.